zebra/interface.c

```diff
@@ -63,22 +63,46 @@ static const char *if_zebra_data_state(uint8_t state)
 static void if_zebra_speed_update(struct event *event)
 {
 	struct interface *ifp = EVENT_ARG(event);
-	struct zebra_if *zif = ifp->info;
+
+	dplane_intf_speed_get(ifp);
+}
+
+static void zebra_if_schedule_speed_update(struct zebra_if *zif, int timeout)
+{
+	if (!CHECK_FLAG(zif->flags, ZIF_FLAG_SPEED_POLLING))
+		return;
+
+	if (event_is_scheduled(zif->speed_update))
```

Comment:
don't need this kind of check - if the task is already scheduled, the 'add_timer' api will check for that under a lock...